macros/relationships/append_only/aggregate_in_before.sql

```diff
@@ -0,0 +1,23 @@
+{% macro aggregate_in_before_join_clause(i) %}
+
+{% set stream = dbt_activity_schema.generate_stream_alias %}
+{% set columns = dbt_activity_schema.columns() %}
+
+(
+    {{ stream(i) }}.{{- columns.ts }} < {{ stream() }}.{{- columns.ts }}
+)
+{% endmacro %}
+
+{% macro aggregate_in_before_aggregation_func() %}
+count(distinct {{ caller() }})
```

Comment:
Caller functionally in jinja allows the macro to act as a decorator. See here for details: https://jinja.palletsprojects.com/en/3.1.x/templates/#call